smack-integration-test/src/main/java/org/igniterealtime/smack/inttest/SmackIntegrationTestFramework.java

```diff
@@ -149,6 +163,22 @@ public static void main(String[] args) throws IOException, KeyManagementExceptio
         System.exit(exitStatus);
     }
 
+    private static String getSpecificationReference(FailedTest failedTest) {
+        final SmackIntegrationTest test = failedTest.concreteTest.method.getAnnotation(SmackIntegrationTest.class);
+        final SpecificationReference spec = failedTest.concreteTest.method.getAnnotation(SpecificationReference.class);
```

Comment:
Mustn't this be retrieved from the class declaring the method?

Also, if not existent, then early exit returning `null`.